rs/replica/src/setup_ic_stack.rs

```diff
@@ -131,6 +131,9 @@ pub fn construct_ic_stack(
         registry.get_latest_version(),
         registry.as_ref(),
     );
+
+    let delegation_from_nns = Arc::new(OnceCell::new());
```

Comment:
Do you really need the Arc here?

Reply:
The `OnceCell` clone implementation creates a new `OnceCell`. In our case we want to share the same `OnceCell` thus the `Arc`.